item-impl/src/main/java/com/example/auction/item/impl/ItemRepository.java

```diff
@@ -168,7 +163,12 @@ public PItemEventProcessor(CassandraSession session, CassandraReadSide readSide)
             );
         }
 
+        private void registerCodec(Session session, TypeCodec<?> codec) {
+            session.getCluster().getConfiguration().getCodecRegistry().register(codec);
+        }
+
         private CompletionStage<Done> prepareStatements() {
+            session.underlying().thenAccept(s -> registerCodec(s, new EnumNameCodec<>(ItemStatus.class)));
```

Comment:
Probably should be added into the `doAll` sequence so that the `CompletionStage` returned by this method awaits this as well.